mono/metadata/threads.c

```diff
@@ -2334,7 +2349,7 @@ mono_thread_current_check_pending_interrupt (void)
 }
 
 static gboolean
-request_thread_abort (MonoInternalThread *thread, MonoObject *state, gboolean appdomain_unload)
+request_thread_abort (MonoInternalThread *thread, MonoObjectHandle *state, gboolean appdomain_unload)
```

Comment:
Because there are unconverted raw paths that pass it as null.
Consider a pointer to a handle to be the union of an unused raw optional pointer, and a handle. I didn't think that was worth a separate function. I considered commenting it but thought it maybe clear. I'll add a comment. I also wasn't sure where comments on parameters go, but I remember now, if I'm willing to write a large function-comment.

Reply:
Ah, I see.  Yea I think it probably warrants a FIXME comment to go back and change it over to just use `MonoObjectHandle state` once all the callers have been updated.

Reply:
Why can't we have the following:
```C
static gboolean
request_thread_abort (MonoInternalThread *thread, MonoObject *state, gboolean appdomain_unload)
{
    return request_thread_abort_handle (thread, MONO_HANDLE_NEW (MonoThreadObject, state), appdomain_unload);
}

static gboolean
request_thread_abort_handle (MonoInternalThread *thread, MonoObjectHandle state, gboolean appdomain_unload)
{
   ...
}
```

Reply:
extra cost -- function enter, allocate handle, function_return.

Reply:
Extra cost doesn't matter, enter/exit function is not needed. Please keep code simple to read. And please remember that we are going away from `MonoObject*` and `MonoHandle` is the new normal, and we shouldn't leave any "mopping" behind (like you said yourself).

Reply:
A pointer to a handle to allow for optionality in non-handle code isn't simple enough?
It isn't obviously safe to allocate the handles in the other callers, that seems a little less simple?
Another idea here is to pass NULL as a handle value, rather than a handle containing NULL. You know the pointer to pointer being null, which is unusual in handle code. We always proactively allocate them.

Reply:
This code has proven very sensitive. The smaller change is preferred.